dvc/repo/__init__.py

```diff
@@ -122,6 +122,33 @@ def _get_repo_dirs(
 
         return root_dir, dvc_dir, tmp_dir
 
+    def _get_subsystem_tmp_dir(
+        self, subsystem, root_dir, tmp_dir, config: dict
+    ):
```

Comment:
How about `_get_database_dir(db_name)`?
`state` and `index` are essentialy databases identified in config by their name. 
I think that `tmp` can be skipped, since from db point of view we don't care if its tmp or not. tmp is a naming convention that can be used by users to deduct what can be removed in project without breaking it severly.